pints/_log_likelihoods.py

```diff
@@ -861,6 +862,323 @@ def evaluateS1(self, x):
+            else:
+                b = pints.vector(b)
+                if len(b) != self._no:
+                    raise ValueError(
+                        'Upper limit must be a ' +
+                        ' scalar or a vector of length n_outputs.')
+
+        if (a is not None) and (b is not None):
+            diff = b - a
+            if np.any(diff <= 0):
+                raise ValueError('Upper limit ' +
+                                 'must exceed lower limit.')
+        self._a = a
+        self._b = b
+
+        # Define the condition for whether a point is not censored
+        if (self._a is not None) and (self._b is not None):
+            self._condition = (self._a < self._values
+                               ) & (self._values < self._b)
+        elif (self._a is not None) and (self._b is None):
+            self._condition = self._a < self._values
+        elif (self._a is None) and (self._b is not None):
+            self._condition = self._values < self._b
+        else:
+            raise ValueError("Not a censored likelihood")
+
+        # Check that none of the values are less than the lower
+        # limit or greater than the upper limit
+        if self._values.ndim == 1:
+            for i in range(np.shape(self._values)[0]):
+                if a is not None:
+                    if self._values[i] < self._a[0]:
+                        raise ValueError("A value is below the lower limit.")
+                if b is not None:
+                    if self._values[i] > self._b[0]:
+                        raise ValueError("A value is above the upper limit.")
+        else:
+            for i in range(np.shape(self._values)[1]):
+                for j in range(np.shape(self._values)[0]):
+                    if a is not None:
+                        if self._values[j, i] < self._a[i]:
+                            raise ValueError(
+                                "A value is below the lower limit.")
+                    if b is not None:
+                        if self._values[j, i] > self._b[i]:
+                            raise ValueError(
+                                "A value is above the upper limit.")
+
+        # Print data and values that are censored
+        if verbose:
+            print("The data are {}. \n The lower censored values"
+                  " are {}. \n The upper censored values"
+                  " are {}.".format(self._values,
+                                    np.extract(self._values == self._a,
+                                               self._values),
+                                    np.extract(self._values == self._b,
+                                               self._values)))
+
+        # Number of points that aren't censored (for each observation
+        # in the multioutput case)
+        self._n_not_censored = np.sum(self._condition, axis=0)
+
+        # Pre-calculate parts
+        self._logn = 0.5 * self._n_not_censored * np.log(2 * np.pi)
```

Comment:
Accuracy comment:

This is quite cheap to calculate and you actually increase the floating point error by pre-calculating it. So, perhaps, you could just move that back into the __call__ and evaluate method.